### PR TITLE
feat(chip): add disabled property

### PR DESCRIPTION
**Pull request by BerkeAras**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`npm run build`) was run locally and any changes were pushed
- [X] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
Chips could not be disabled

Issue Number: #19510 

## What is the new behavior?
Chips can now be disabled

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: core/src/components/chip/chip.tsx [hunk 1]
```diff
@@ -28,16 +28,25 @@ export class Chip implements ComponentInterface {
...
+  @Prop({ reflectToAttr: true }) disabled = false;
+
   render() {
     const mode = getIonMode(this);
 
+    const { disabled } = this;
+
     return (
       <Host
+     aria-disabled={disabled ? 'true' : null}
```
Comment (larsblumberg) [inline core/src/components/chip/chip.tsx:43]: broken indent

```suggestion
        aria-disabled={disabled ? 'true' : null}
```

File: core/src/components/chip/chip.tsx [hunk 2]
```diff
@@ -28,16 +28,25 @@ export class Chip implements ComponentInterface {
...
         class={{
           ...createColorClasses(this.color),
           [mode]: true,
           'chip-outline': this.outline,
-          'ion-activatable': true,
+            'ion-activatable': true,
+            'chip-disabled': disabled,
```
Comment (larsblumberg) [inline core/src/components/chip/chip.tsx:49]:
```suggestion
          'ion-activatable': true,
          'chip-disabled': disabled,
```

File: core/src/components/chip/test/basic/index.html [hunk 3]
```diff
@@ -49,7 +49,10 @@ <h2>Basic Chips</h2>
         </ion-avatar>
         <ion-label>With Icon and Avatar</ion-label>
         <ion-icon name="close-circle"></ion-icon>
-      </ion-chip>
+	 </ion-chip>
+	 <ion-chip disabled>
+		<ion-label>Disabled</ion-label>
+	 </ion-chip>
```
Comment (larsblumberg) [inline core/src/components/chip/test/basic/index.html:55]: broken indent

Comment (larsblumberg) [reply core/src/components/chip/test/basic/index.html:55]: And also all the changes below ;)

Comment (BerkeAras) [reply core/src/components/chip/test/basic/index.html:55]: Thank you @larsblumberg

**Comment by HansKrywaa** [comment]
There is already a PR for this: #20658

**Pull request closed without merging**